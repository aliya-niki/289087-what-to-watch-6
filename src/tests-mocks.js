export const filmsRaw = [{
  "id": 1,
  "name": `The Grand Budapest Hotel`,
  "preview_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  "rating": 8.9,
  "scores_count": 240,
  "director": `Wes Andreson`,
  "starring": [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`],
  "run_time": 99,
  "genre": `Comedy`,
  "released": 2014,
  "is_favorite": true
}, {
  "id": 2,
  "name": `Bohemian Rhapsody`,
  "preview_image": `img/bohemian-rhapsody.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `Freddie Mercury -- the lead singer of Queen -- defies stereotypes and convention to become one of history's most beloved entertainers.`,
  "rating": 8.5,
  "scores_count": 240,
  "director": `Bryan Singer`,
  "starring": [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  "run_time": 134,
  "genre": [`Drama`, `Biography`],
  "released": 2018,
  "is_favorite": true
}, {
  "id": 3,
  "name": `Macbeth`,
  "preview_image": `img/macbeth.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `After a brilliant victory in battle, Macbeth returns to an indolent court where honor is dispensed by whim.`,
  "rating": 7.9,
  "scores_count": 240,
  "director": `Kit Monkman`,
  "starring": [`Mark Rowley`, `Al Weaver`, `David Bradley`],
  "run_time": 121,
  "genre": `Drama`,
  "released": 2018,
  "is_favorite": false
}];

export const filmsAdapted = [{
  id: 1,
  name: `The Grand Budapest Hotel`,
  previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: true
}, {
  id: 2,
  name: `Bohemian Rhapsody`,
  previewImage: `img/bohemian-rhapsody.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Freddie Mercury -- the lead singer of Queen -- defies stereotypes and convention to become one of history's most beloved entertainers.`,
  rating: 8.5,
  scoresCount: 240,
  director: `Bryan Singer`,
  starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  runTime: 134,
  genre: [`Drama`, `Biography`],
  released: 2018,
  isFavorite: true
}, {
  id: 3,
  name: `Macbeth`,
  previewImage: `img/macbeth.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `After a brilliant victory in battle, Macbeth returns to an indolent court where honor is dispensed by whim.`,
  rating: 7.9,
  scoresCount: 240,
  director: `Kit Monkman`,
  starring: [`Mark Rowley`, `Al Weaver`, `David Bradley`],
  runTime: 121,
  genre: `Drama`,
  released: 2018,
  isFavorite: false
}];

export const authRaw = {
  "id": 1,
  "email": `keks@mail.com`,
  "name": `keks`,
  "avatar_url": `keks.jpeg`,
};

export const authAdapted = {
  id: 1,
  email: `keks@mail.com`,
  name: `keks`,
  avatarUrl: `keks.jpeg`,
};

export const user = {
  email: `keks@mail.com`,
  password: 12345678
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 111,
      name: `Kate Fogler`
    },
    rating: 9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 2,
    user: {
      id: 287,
      name: `Kate Muir`
    },
    rating: 6,
    comment: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 3,
    user: {
      id: 17,
      name: `Kate Hidi`
    },
    rating: 2,
    comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 4,
    user: {
      id: 43,
      name: `Kate Snow`
    },
    rating: 4,
    comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
