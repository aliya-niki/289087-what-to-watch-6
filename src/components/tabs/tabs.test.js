import React from 'react';
import {render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {filmsAdapted} from '../../tests-mocks';
import Tabs, {MovieCardTabs} from './tabs';

const mockFilm = filmsAdapted[2];

describe(`Tabs`, () => {
  it(`should render correctly with active 'Overview' tab`, () => {
    const {container} = render(
        <Tabs film={mockFilm}/>
    );

    expect(screen.getByText(mockFilm.description)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it(`should render correctly with active 'Details' tab`, () => {
    const {container} = render(
        <Tabs film={mockFilm}/>
    );

    userEvent.click(screen.getByTestId(MovieCardTabs.DETAILS));
    expect(screen.getByText(mockFilm.genre)).toBeInTheDocument();
    expect(screen.getByText(mockFilm.director)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
