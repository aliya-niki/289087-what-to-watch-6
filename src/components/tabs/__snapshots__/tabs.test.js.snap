// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly with active 'Details' tab 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            data-testid="Overview"
            data-value="Overview"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            data-testid="Details"
            data-value="Details"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            data-testid="Reviews"
            data-value="Reviews"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-card__text movie-card__row"
    >
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Director
          </strong>
          <span
            class="movie-card__details-value"
            style="white-space: pre;"
          >
            Kit Monkman
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Starring
          </strong>
          <span
            class="movie-card__details-value"
            style="white-space: pre;"
          >
            Mark Rowley, 
Al Weaver, 
David Bradley
          </span>
        </p>
      </div>
      <div
        class="movie-card__text-col"
      >
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Run Time
          </strong>
          <span
            class="movie-card__details-value"
          >
            2h 1m
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Genre
          </strong>
          <span
            class="movie-card__details-value"
            style="white-space: pre;"
          >
            Drama
          </span>
        </p>
        <p
          class="movie-card__details-item"
        >
          <strong
            class="movie-card__details-name"
          >
            Released
          </strong>
          <span
            class="movie-card__details-value"
          >
            2018
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly with active 'Overview' tab 1`] = `
<div>
  <div
    class="movie-card__desc"
  >
    <nav
      class="movie-nav movie-card__nav"
    >
      <ul
        class="movie-nav__list"
      >
        <li
          class="movie-nav__item movie-nav__item--active"
        >
          <a
            class="movie-nav__link"
            data-testid="Overview"
            data-value="Overview"
            href="#"
          >
            Overview
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            data-testid="Details"
            data-value="Details"
            href="#"
          >
            Details
          </a>
        </li>
        <li
          class="movie-nav__item "
        >
          <a
            class="movie-nav__link"
            data-testid="Reviews"
            data-value="Reviews"
            href="#"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="movie-rating"
    >
      <div
        class="movie-rating__score"
      >
        7.9
      </div>
      <p
        class="movie-rating__meta"
      >
        <span
          class="movie-rating__level"
        >
          Good
        </span>
        <span
          class="movie-rating__count"
        >
          240
           ratings
        </span>
      </p>
    </div>
    <div
      class="movie-card__text"
    >
      <p>
        After a brilliant victory in battle, Macbeth returns to an indolent court where honor is dispensed by whim.
      </p>
      <p
        class="movie-card__director"
      >
        <strong>
          Director: 
          Kit Monkman
        </strong>
      </p>
      <p
        class="movie-card__starring"
      >
        <strong>
          Starring: 
          Mark Rowley, Al Weaver, David Bradley
           and other
        </strong>
      </p>
    </div>
  </div>
</div>
`;
